src/main/java/com/github/princesslana/greedorama/commands/HistoryCommand.java

```diff
@@ -0,0 +1,92 @@
+package com.github.princesslana.greedorama.commands;
+
+import com.github.princesslana.greedorama.PortfolioRepository;
+import com.github.princesslana.greedorama.Transaction;
+import com.github.princesslana.greedorama.UserRepository;
+import com.google.common.base.Joiner;
+import com.google.common.collect.Ordering;
+import disparse.discord.smalld.DiscordRequest;
+import disparse.discord.smalld.DiscordResponse;
+import disparse.parser.reflection.CommandHandler;
+import disparse.parser.reflection.Flag;
+import disparse.parser.reflection.ParsedEntity;
+import java.util.HashSet;
+
+public class HistoryCommand {
+
+  private static final int TXNS_PER_PAGE = 10;
+
+  private final DiscordRequest request;
+
+  private final PortfolioRepository portfolios;
+
+  private final UserRepository users;
+
+  public HistoryCommand(
+      DiscordRequest request, PortfolioRepository portfolios, UserRepository users) {
+    this.request = request;
+    this.portfolios = portfolios;
+    this.users = users;
+  }
+
+  @ParsedEntity
+  private static class Options {
+    @Flag(shortName = 'p', longName = "page", description = "Display page")
+    public Integer page = 1;
+  }
+
+  @CommandHandler(commandName = "history")
+  public DiscordResponse portfolio(Options options) {
+    var guildId = request.getDispatcher().guildFromEvent(request.getEvent());
+    var userId = request.getDispatcher().identityFromEvent(request.getEvent());
+
+    var user = users.get(guildId, userId);
+    var portfolio = portfolios.get(user);
+
+    var txns = portfolio.getTransactions();
+    var totalPages = (txns.size() + TXNS_PER_PAGE - 1) / TXNS_PER_PAGE;
+    var page = options.page;
+
+    var pageText = String.format("Page %d of %d", options.page, totalPages);
+
+    if (options.page > totalPages || options.page <= 0) {
+      return Format.error("Page number out of range");
+    }
+
+    var dates = new HashSet<String>();
+    var txnList = new StringBuilder();
+
+    var byWhen = Ordering.natural().reverse().onResultOf(Transaction::getWhen);
+    txns = byWhen.sortedCopy(txns);
```

Comment:
Aren't the transactions returned by Portfolio already sorted?

Reply:
They're sorted oldest to newest, whereas I feel that the most recent transactions should pop up first. I could also just iterate through the list backwards, if that would be more efficient.

Reply:
You're right, most recent should pop up first. What I like to do here is make a `Transactions.byWhen` method that returns an Ordering so we don't have to duplicate this. There you would then do `Transactions.byWhen().reverse().sortedCopy(txns)`. Doesn't really save code, but it keeps the idea centralized.

Reply:
Okay. Should I also apply the same idea with the `byNetWorth` ordering in `PortfolioCommand` with a `Portfolio.byNetWorth`?

Reply:
It can be, yeah. If we're only using an ordering once it's not a must do, but it can be. Once we're using an ordering twice I think definitely.

Reply:
Got it, I'll just go ahead and do both for now.